### PR TITLE
feat(onboarding): Add validation support for GCP CDR onboarding

### PR DESCRIPTION
**Pull request by ravinadhruve10**
Change summary:
-----------------
1. Support for GCP CDR validation during onboarding using WIF. Mainly fixed the case for metadata population when WIF is passed.
2. Added GCP CDR tests.
3. Modified the structure and organization of acceptance test suite for sysdig_secure_cloud_auth_account resource.

Testing done:
---------------
- Validated this end to end using local TF sysdig provider plugin on staging.
- Validated the acceptance tests against staging.

### CODE REVIEW

File: sysdig/resource_sysdig_secure_cloud_auth_account_test.go [hunk 1]
```diff
@@ -107,7 +110,210 @@ resource "sysdig_secure_cloud_auth_account" "sample-1" {
...
+			{
+				ResourceName:      "sysdig_secure_cloud_auth_account.gcp_agentless_scanning",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+// XXX: add project_number to this test
```
Comment (ravinadhruve10) [inline sysdig/resource_sysdig_secure_cloud_auth_account_test.go:253]: Pushing a backend fix for this. However, this test will require the proto fix in all envs to pass.

Comment (ravinadhruve10) [reply sysdig/resource_sysdig_secure_cloud_auth_account_test.go:253]: Fix for bumping up proto spec merged to staging. Once latest cloudauth gets released to all regions will add back the `project_number` field in the test. Till then this PR is ready to be reviewed.

File: sysdig/resource_sysdig_secure_cloud_auth_account.go [hunk 2]
```diff
@@ -387,25 +388,28 @@ func constructAccountComponents(data *schema.ResourceData) []*cloudauth.AccountC
...
-								ServicePrincipalMetadata: &cloudauth.ServicePrincipalMetadata{
-									Provider: &cloudauth.ServicePrincipalMetadata_Gcp{
-										Gcp: &cloudauth.ServicePrincipalMetadata_GCP{
-											Key:                        spGcpKey,
-											WorkloadIdentityFederation: spGcp.Gcp.WorkloadIdentityFederation,
-											Email:                      spGcp.Gcp.Email,
-										},
+							err = json.Unmarshal(spGcpKeyBytes, &spGcpKey)
+						}
+						log.Printf("Setting key: %v ; wif: %v ; email: %v", spGcpKey, spGcp.Gcp.WorkloadIdentityFederation, spGcp.Gcp.Email)
```
Comment (mdkulkarni15) [inline sysdig/resource_sysdig_secure_cloud_auth_account.go:402]: Do you want to log the key here?

File: sysdig/resource_sysdig_secure_cloud_auth_account_test.go (hunk 1 above)
Comment (mdkulkarni15) [reply sysdig/resource_sysdig_secure_cloud_auth_account_test.go:253]: I thought we don't need to take project_number as an input, unless you are talking about something else.

File: sysdig/resource_sysdig_secure_cloud_auth_account.go (hunk 2 above)
Comment (ravinadhruve10) [reply sysdig/resource_sysdig_secure_cloud_auth_account.go]: Oops, thanks for pointing this out. Fixing it in next commit.

File: sysdig/resource_sysdig_secure_cloud_auth_account_test.go (hunk 1 above)
Comment (ravinadhruve10) [reply sysdig/resource_sysdig_secure_cloud_auth_account_test.go:253]: Actually, the final decision we took was to take it as input and added the same in proto spec.

Comment (ravinadhruve10) [reply sysdig/resource_sysdig_secure_cloud_auth_account_test.go]: Done and updated the test.

**Comment by ravinadhruve10** [comment]
The CI test failures are due to other tests :- 
--- FAIL: TestAccManagedRuleset (3.66s)
--- FAIL: TestAccManagedPolicy (2.86s)

The added test in this PR is passing infact :- 
--- PASS: TestAccGCPSecureCloudAuthAccountThreatDetection (2.56s)

**Review by mdkulkarni15 — APPROVED** [review]

**Review by haresh-suresh — APPROVED** [review]

**Pull request merged**